shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -626,15 +626,23 @@ class UCX(transport: UCXShuffleTransport, executor: BlockManagerId, rapidsConf:
     endpointManager.getConnection(peerExecutorId, peerMgmtHost, peerMgmtPort)
   }
 
-  def onWorkerThreadAsync(task: () => Unit): Unit = {
-    workerTasks.add(task)
+  private def signalWorker(): Unit = {
     if (rapidsConf.shuffleUcxUseWakeup) {
       withResource(new NvtxRange("UCX Signal", NvtxColor.RED)) { _ =>
-        worker.signal()
+        try {
+          worker.signal()
+        } catch {
+          case e: Throwable if isShuttingDown => // ignore these
```

Comment:
Normally I would expect an NPE to be fixed by adding some check somewhere to make sure we don't deference something that can be null.  I'm normally not a fan of fixing NPE's by catch-alls and ignoring, as it can submarine all sorts of other issues accidentally.

What was the cause of the NPE?  Is it fixed or are we just catching and ignoring as a "fix"?

Reply:
The NPEs are caused by double-closing in UCX. We close once, and a nativeId `Long` is set to `null`, but then if there's a second call to close, the `null` is normally sent to a JNI call which expects `long` primitives. This is producing a `NullPointerException`, as java can't convert the `Long` null to `long`.

So that's what I am trying to fix, but you are right it could be done better. Given that `signal` can be called from other threads (literally anyone trying to queue up a send or a receive ends up there) and `close` is called from the progress thread when exiting, we really need a lock between the `close` and `signal` functions for `worker`.

Reply:
@jlowe, pushed this (https://github.com/NVIDIA/spark-rapids/pull/2950/commits/e921d038adf86eaf4e666b2647ae8472bd3b9739) to handle the above.